### ISSUE TITLE
Use LDAP user name instead of UUID for federated cloud id

### ISSUE THREAD

**Issue by patrick1990** [nextcloud/server#5406]
Nextcloud (up to version 12.0.0) generates federated cloud ids of the shape uid@server. This leads to inconvenient cloud ids in case of LDAP users because the LDAP user app configures the UUID as Nextcloud uid per default. This is changeable via the LDAP configuration panel but changes only behaviour for new users (because I cannot clear the LDAP user mapping in a production environment).

It would be nice if this would be configurable on the federation cloud id level to generate cloud ids for LDAP users from the LDAP username.

Maybe the new cloud id api from pull request #3297 can be handy to implement this feature?

(s. also Nextcloud Forum here: https://help.nextcloud.com/t/changing-the-internal-user-name/8924)

**Comment by blizzz** [comment]
@schiessle perhaps we should just show the federated id based on the last-used loginname instead of the username. Latter isn't considered for anything else than internal usage anyway… only problem: login name isn't necessarily unique.

**Comment by patrick1990** [comment]
Please also notice that the federated cloud id is also shown to the user of the remote Nextcloud who receives the share. It will be hard to associate the share to a person that you know if you read that the share comes from "12345678-abcd-9827-123456789abcd@cloud.example.com".

**Comment by dsommers** [comment]
I've linked Nextcloud against a FreeIPA server.  In the LDAP settings, in the Expert tab, there is an `Internal Username Attribute:` field.  In my case, I set this to `uid`.  And that gives me the normal IPA usernames in Nextcloud as well, and makes the federated ID `username@cloud.example.com`

**Comment by patrick1990** [comment]
@dsommers yes, but this configuration only changes the username for new users. To change the uid for old users I would have to delete the LDAP user mapping what would destroy my production environment.

In fact I don't really care about the internal username that Nextcloud and the LDAP app are using. I think a convinient solution would be to link internal usernames used in the Nextcloud database to names shown to the users in the web interface. When I have understood #3297 correctly, this is exactly what the new CloudID interface was made for.

**Comment by Goldenflamer** [comment]
I have the same issue, would be great if this could be improved.

**Comment by NathanielHill** [comment]
Same issue, I would simply like to be able to remove the subdomain so the IDs are user@mydomain.com instead of user@cloud.mydomain.com

**Comment by blizzz** [comment]
@NathanielHill that won't help you. Your Nextcloud would need to be available at mydomain.com. That's independend from the user part before the @.

**Comment by mvvvmd** [comment]
We have the same issue in Nextcloud 17, just added a server and it looks very confusing to users when receiving a remote share notification.

**Comment by itmanagerro** [comment]
Same for the .mobileconfig file for Calendar... it tries to login via uuid + password, but the user expects to use user + password 
#29433

**Comment by slu-010101** [comment]
We running into the same issue, federated cloud id with username will be awesome.
